src/transform.js

```diff
@@ -99,39 +102,56 @@ export default class i18nTransform extends Transform {
   }
 
   _flush(done) {
-    let catalog = {}
-
     if (this.options.sort) {
       this.entries = this.entries.sort((a, b) => a.key.localeCompare(b.key))
     }
 
-    let uniqueCount = this.entries.length
-    for (const entry of this.entries) {
-      const { duplicate, conflict } = dotPathToHash(
-        entry,
-        catalog,
-        {
-          separator: this.options.keySeparator,
-          value: this.options.defaultValue,
-          useKeysAsDefaultValue: this.options.useKeysAsDefaultValue
-        }
-      )
-      if (duplicate) {
-        uniqueCount -= 1
-        if (conflict) {
-          const warning = `Found same keys with different values: ${entry.key}`
-          this.emit('warning', warning)
-          if (this.options.verbose) {
-            warn(warning)
+    for (const locale of this.options.locales) {
+      const catalog = {}
+      const { numbers } = i18next.services.pluralResolver.getRule(locale)
+
+      let countWithPlurals = 0
+      let uniqueCount = this.entries.length
+
+      const transformEntry = (entry, suffix) => {
+        const { duplicate, conflict } = dotPathToHash(
+          entry,
+          catalog,
+          {
+            suffix,
+            separator: this.options.keySeparator,
+            value: this.options.defaultValue,
+            useKeysAsDefaultValue: this.options.useKeysAsDefaultValue
           }
+        )
+
+        if (duplicate) {
+          uniqueCount -= 1
+          if (conflict) {
+            const warning = `Found same keys with different values: ${entry.key}`
+            this.emit('warning', warning)
+            if (this.options.verbose) {
+              warn(warning)
+            }
+          }
+        } else {
+          countWithPlurals += 1
+        }
+      }
+
+      for (const entry of this.entries) {
+        if (typeof entry.count !== 'undefined') {
+          numbers.forEach((_, i) => {
+            transformEntry(
+              entry,
+              numbers.length > 2 ? i : i ? 'plural' : ''
```

Comment:
I'm a bit reluctant to use of ternary operators in general. I didn't comment [above](https://github.com/i18next/i18next-parser/pull/191/files#diff-b38f7254be1a9f817fd2e33c294481a1R10) because if was a fairly simple one and still very readable. This one is total chinese to me and the intent behind it should be capture in as many loc as necessary. I prefer something like this:
```js
const isMeetingTestBla = numbers.length > 2
let suffix
if (isMeetingTestBla) {
  suffix = i
}
else if (i) {
  suffix = 'plural'
}
else {
  suffix = ''
}
```